iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/event_listener.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+#ifndef IOX_POSH_POPO_BUILDING_BLOCKS_EVENT_LISTENER_HPP
+#define IOX_POSH_POPO_BUILDING_BLOCKS_EVENT_LISTENER_HPP
+
+#include "iceoryx_posh/internal/popo/building_blocks/event_variable_data.hpp"
+#include "iceoryx_utils/cxx/helplets.hpp"
+
+namespace iox
+{
+namespace popo
+{
+/// @brief An EventListener performs a blocking wait on a shared event variable.
+///        When wait returns a list of all the EventNotifier id's which had
+///        triggered the EventVariable is returned and the state is reset.
+///
+/// @attention Do not use multiple EventListener at the same time for the same EventVariable
+class EventListener
+{
+  public:
+    /// @brief creates new EventListener
+    ///
+    /// @param[in] dataRef reference to EventVariableData
+    EventListener(EventVariableData& dataRef) noexcept;
+
+    /// @brief returns vector of indices of active notifications; blocking if EventVariableData was
+    /// not notified unless destroy() was called before. The indices of active notifications is
+    /// never empty unless destroy() was called, then it's always empty.
+    ///
+    /// @return vector of active notifications
+    cxx::vector<cxx::BestFittingType_t<MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET>,
+                MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET>
```

Comment:
Please create a type alias with a `using` declaration, e.g. `NotificationVector_t` like it is already done in `test_popo_event_variable.cpp`. This is used at multiple places and non-trivial.